src/app/mesh/qgsmaptooleditmeshframe.cpp

```diff
@@ -269,6 +274,26 @@ QgsMapToolEditMeshFrame::QgsMapToolEditMeshFrame( QgsMapCanvas *canvas )
       mSelectionBand->reset( Qgis::GeometryType::Polygon );
   } );
 
+  connect( mActionSelectIsolatedVertices, &QAction::triggered, this, [this]
+  {
+    onEditingStarted();
+    setSelectedVertices( mCurrentEditor->freeVerticesIndexes(), Qgis::SelectBehavior::SetSelection );
+  } );
+
+  connect( mActionSelectAllVertices, &QAction::triggered, this, [this]
+  {
+    onEditingStarted();
+
+    QList<int> verticesIndexes;
+    verticesIndexes.reserve( mCurrentLayer->meshVertexCount() );
+    for ( int i = 0; i < mCurrentLayer->meshVertexCount(); i++ )
+    {
+      verticesIndexes.append( i );
+    }
```

Comment:
`meshVertexCount()` says:

> \note during mesh editing, some vertices can be void and are not included in this returned value

does that affect us?
Would it be better to maybe use `mCurrentLayer->selectVerticesByExpression( true )` to get the indexes instead?

Reply:
Seems like a good point.

We could avoid evaluating the expression by using `mCurrentLayer->nativeMesh()->vertexCount()`. That would skip need to use `QgsExpression` and its evaluation. It is what the `mCurrentLayer->selectVerticesByExpression()` does internally anyways.